### PR TITLE
fixed error in reduced argument

### PR DESCRIPTION
**Pull request by Robaina**
The method run_deseq() gave an error whenever the argument "reduced" was used. This argument requires a formula, just like "design_matrix", however, the code didn't do the conversion to the Formula object. This pull request fixes the error.

### CODE REVIEW

**Comment by wckdouglas** [comment]
Hi @Robaina, thank you for fixing the problem! Do you mind also adding a test in [test_deseq.py](https://github.com/wckdouglas/diffexpr/blob/master/test/test_deseq.py) (and maybe [deseq.R](https://github.com/wckdouglas/diffexpr/blob/master/test/deseq.R)) to make sure future changes won't break this? Let me know if you're unsure how to do it, I can add that too! Thank you again for the PR!

**Comment by Robaina** [comment]
> Hi @Robaina, thank you for fixing the problem! Do you mind also adding a test in [test_deseq.py](https://github.com/wckdouglas/diffexpr/blob/master/test/test_deseq.py) (and maybe [deseq.R](https://github.com/wckdouglas/diffexpr/blob/master/test/deseq.R)) to make sure future changes won't break this? Let me know if you're unsure how to do it, I can add that too! Thank you again for the PR!

Hi, no problem, I could do that throughout the week

File: diffexpr/py_deseq.py [hunk 1]
```diff
@@ -92,6 +92,10 @@ def run_deseq(self, **kwargs):
             BPPARAM = bpparam()
         )
         """
+        
+        for arg in kwargs.keys():
+            if arg == 'reduced':
+                kwargs[arg] = Formula(kwargs[arg])
```
Comment (wckdouglas) [inline diffexpr/py_deseq.py]:
```suggestion
        for key, value in kwargs.items():
            if key == 'reduced':
                kwargs[key] = Formula(value)
```

just a minor suggestion, but I think using ```.item``` is cleaner when we modify a dictionary

Comment (Robaina) [reply diffexpr/py_deseq.py]: Alright :)

**Pull request merged**